// @easy-webpack/config-aurelia/dist/index.js
"use strict";
var core_1 = require('@easy-webpack/core');
var AureliaWebpackPlugin = require('aurelia-webpack-plugin');
module.exports = function aurelia(_a) {
    var _b = _a === void 0 ? {} : _a, _c = _b.root, root = _c === void 0 ? '' : _c, _d = _b.src, src = _d === void 0 ? '' : _d, _e = _b.title, title = _e === void 0 ? 'Aurelia' : _e, _f = _b.baseUrl, baseUrl = _f === void 0 ? '/' : _f;
    var allOptions = arguments[0];
    return function aurelia() {
        return {
            metadata: {
                title: title,
                baseUrl: baseUrl,
                root: root,
                src: src,
            },
            resolve: {
                root: src,
                modules: [src].concat(core_1.get(this, 'resolve.modules', ['node_modules']))
            },
            plugins: [
                new AureliaWebpackPlugin(allOptions)
            ].concat(core_1.get(this, 'plugins', []))
        };
    };
};
/*
export function useAureliaES2015(aliases: { [packageName: string]: string }) {
  return function aureliaES2015(this: WebpackConfig): WebpackConfig {
    return {
      resolve: {
        alias: aliases
      }
    }
  }
}
*/
